File: src/main/java/com/team/puddy/global/config/QuerydslConfig.java
package com.team.puddy.global.config;

import com.querydsl.jpa.impl.JPAQueryFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.persistence.EntityManager;

@Configuration
public class QuerydslConfig {
    @Bean
    JPAQueryFactory jpaQueryFactory(EntityManager entityManager) {
        return new JPAQueryFactory(entityManager);
    }
}

File: src/main/kotlin/com/team/puddy/global/config/querydsl/QuerydslConfig.kt
package com.team.puddy.global.config.querydsl

import com.querydsl.jpa.impl.JPAQueryFactory
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class QuerydslConfig {

    @PersistenceContext
    private lateinit var em: EntityManager

    @Bean
    fun init() = JPAQueryFactory(em)

}
